flow/BlobCipher.cpp

```diff
@@ -133,57 +133,43 @@ BlobCipherKeyIdCacheKey BlobCipherKeyIdCache::getCacheKey(const EncryptCipherBas
 }
 
 Reference<BlobCipherKey> BlobCipherKeyIdCache::getLatestCipherKey() {
-	if (keyIdCache.empty()) {
-		// Cache is empty, nothing more to do.
-		throw encrypt_key_not_found();
+	// ENCRYPT_INVALID_CIPHER_KEY_ID indicate there was no key inserted. In this case we return empty key instead of
+	// throwing encrypt_invalid_id.
+	if (latestBaseCipherKeyId == ENCRYPT_INVALID_CIPHER_KEY_ID) {
```

Comment:
@sfc-gh-ahusain I'm still making change since I want to make it consistent that things isn't actually an error don't throw an exception. Or if you prefer to, I can make `latestBaseCipherKeyId` an `Optional`, and testing against `Optional::present()` rather than whether the id is invalid to tell if the cache is empty.

Reply:
I think you need to rebase the branch with the latest main, I updated this code as part of re-merging the auth header changes. API throws encyrpt_key_not_found if cache is empty; otherwise, it asserts that member values are valid.

Reply:
yeah I saw that change. I'm just so tempting to do without exception :/

Reply:
My suggestion was to pull the change and replace exception with "empty reference"; I thought we agreed that "invalid_id" will be thrown and if key is not present due to: empty cache and/or key_not_found, then "empty reference" will be returned ?

Reply:
updated to use Optional. does it look a little bit better?